src/index/CompressedRelation.cpp

```diff
@@ -278,15 +278,13 @@ CompressedRelationReader::IdTableGenerator CompressedRelationReader::lazyScan(
   // value for some sanity checks at the end of the function.
   const auto originalLimit = limitOffset;
 
-  // Compute the sequence of relevant blocks. If the sequence is empty, there
-  // is nothing to yield.
-  auto relevantBlocks = getRelevantBlocks(scanSpec, blockMetadata);
-  if (ql::ranges::empty(relevantBlocks)) {
+  if (relevantBlockMetadata.empty()) {
```

Comment:
Is it safe to assume that we have correctly prefiltered the `relevant` block metadata? or is it hard to implement a variant of `getRelevantBlocks` that can (often redundantly) adjust the blocks again?